activeweb/src/main/java/org/javalite/activeweb/SessionFacade.java

```diff
@@ -17,39 +17,47 @@
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpSession;
-import java.io.Serializable;
 import java.util.*;
 
 /**
  * Facade to HTTP session. 
  *
  * @author Igor Polevoy
  */
-public class SessionFacade implements Map {
+public class SessionFacade implements Map<String,Object> {
+
+    private static final String[] EMPTY_ARRAY = new String[0];
+
+    private HttpSession httpSession() {
+        HttpServletRequest httpRequest = RequestContext.getHttpRequest();
+        return httpRequest == null ? null : httpRequest.getSession(false);
+    }
+
+    public boolean isExists() {
+        return httpSession() != null;
+    }
+
 
     /**
      * Returns a session ID from underlying session.
      *
      * @return a session ID from underlying session, or null if session does not exist.
      */
     public String id(){
-        HttpServletRequest r = RequestContext.getHttpRequest();
-        if(r == null){
-            return null;
-        }
-        HttpSession session = r.getSession(false);
+        HttpSession session = httpSession();
         return session == null ? null : session.getId();
     }
 
-
     /**
      * Retrieve object from session.
      *
      * @param name name of object.
-     * @return named object. 
+     * @return named object.
      */
-    public Object get(String name){
-        return RequestContext.getHttpRequest().getSession(true).getAttribute(name);
+    @Override
+    public Object get(Object name) {
```

Comment:
this is a Map<String, Object> please revert the key type

Reply:
The parent get () method requires Object as a parameter. (@Override rule)
Declaration from java.util.Map
`
    V get(Object key);
`